uqcsbot/scripts/ascii.py

```diff
@@ -0,0 +1,113 @@
+from uqcsbot import bot, Command
+from requests import get
+from requests.exceptions import RequestException
+from uqcsbot.utils.command_utils import loading_status
+import random
+
+NO_QUERY_MESSAGE = "Can't ASCIIfy nothing... try `!asciify <TEXT>`"
+BOTH_OPTIONS_MESSAGE = "Font can only be random OR specified"
+ERROR_MESSAGE = "Trouble with HTTP Request, can't ASCIIfy :("
+
+ASCII_URL = "http://artii.herokuapp.com/make?text="
+FONT_URL = "http://artii.herokuapp.com/fonts_list"
+
+@bot.on_command("asciify")
+@loading_status
+def handle_asciify(command: Command):
+    """
+    `!asciify [--fontslist] [--randomfont | --<CUSTOM FONT>] <TEXT>` - Returns
+    ASCIIfyed text. `--fontslist` also returns a URL to available fonts, `--randomfont`
+    returns, well... a random font. A custom font from the fonts list can also be
+    specified.
+    """
+    # Makes sure the query is not empty
+    if not command.has_arg():
+        bot.post_message(command.channel_id, NO_QUERY_MESSAGE)
+        return
+    command_args = command.arg.split()
+    random_font = False
+    custom_font = False
+    return_fonts = False
+
+    #check for font list option
+    if '--fontslist' in command_args:
+        return_fonts = True
+        command_args.remove('--fontslist')
+    #check for custom font option
+    fontslist = get_fontslist()
+    if not fontslist:
+        bot.post_message(command.channel_id, ERROR_MESSAGE)
+        return
+    for i in fontslist:
```

Comment:
there's going to be a lot more fonts than command args so you may as well loop over command args instead of fonts (and break out of the loop once they don't start with `--`)

note that if you do this you need to check for `--randomfont` first